### DISCUSSION TITLE
Rhino returns a JavaScript object for certain Java classes

### DISCUSSION THREAD

**Discussion by StSchnell** [mozilla/rhino#1450] [Q&A]
For some time now, I have been dealing with the question why Rhino returns a JavaScript object for certain Java classes.

Here an example:

```javascript
// cx is Java object org.mozilla.javascript.Context
var cx = org.mozilla.javascript.Context.getCurrentContext();
java.lang.System.out.println(typeof cx);
java.lang.System.out.println(cx.getClass().getName());

// sealedSharedScope is JavaScript Object
var sealedSharedScope = cx.initStandardObjects(null, true);
java.lang.System.out.println(typeof sealedSharedScope);
// java.lang.System.out.println(sealedSharedScope.getClass().getName());
java.lang.System.out.println(sealedSharedScope.constructor.name);
```

The first sequence delivers with the variable cx a Java object.
The second sequence delivers with the variable sealedSharedScope a JavaScript object.

```
object
org.mozilla.javascript.Context
object
Object
```

But I expect in the second sequence a Java class org.mozilla.javascript.Context.

What is the reason that a JavaScript object is returned instead of a Java class?
How can I get a Java class, so that I can use its methods and attributes in the following code?

**Comment by tonygermano** [comment]
Every object is technically a java class. If the class implements the `org.mozilla.javascript.Scriptable` interface, then it will appear to the javascript runtime as a javascript object. If it does not implement Scriptable, then the runtime will see it as a wrapped java class.

Context does not implement Scriptable, but the return value of initStandardObjects does.
